### PR TITLE
Enforce positional `artifact_path` for `log_model` in **this repository**

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13268?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13268/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13268
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

This PR might be a little controversial. In MLflow 3.0, `artifact_path` will be **renamed** to `name` (breaking change). Currently, examples/tests depending on `log_model(..., artifact_path="<artifact_path>")` are failing on the `mlflow-3` branch. To make them run, this PR updates `log_model(..., artifact_path="<artifact_path>")` (keyword form) to `log_model(..., "<artifact_path>")` (positional form), and adds a lint rule to enforce it.

```python
# only works on either master or mlflow-3 branch.
mlflow.xxx.log_model(model, artifact_path="<artifact_path>")
mlflow.xxx.log_model(model, name="<artifact_path>")

# works on both master and mlflow-3
mlflow.xxx.log_model(model, "<artifact_path>")
```

#### Why not rename `artifact_path` to `name` on the `mlflow-3` branch?

I would if the number of log_model calls that need to be updated was much smaller. Updating all `log_model` calls on the `mlflow-3` branch would increase chances of conflicts.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/pyfunc/test_virtualenv.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def temp_mlflow_env_root(tmp_path, monkeypatch):
 @use_temp_mlflow_env_root
 def test_restore_environment_with_virtualenv(sklearn_model):
     with mlflow.start_run():
-        model_info = mlflow.sklearn.log_model(sklearn_model.model, artifact_path="model")
+        model_info = mlflow.sklearn.log_model(sklearn_model.model, "model")
```
Comment (harupy) [inline tests/pyfunc/test_virtualenv.py:68]: All `log_model` calls specifying `artifact_path` as a positional argument will be transformed by this PR.

**Comment by serena-ruan** [comment]
@harupy Would this break more customers if they are using keyword arguments now?

**Comment by harupy** [comment]
This PR doesn't have any user impacts.

**Comment by serena-ruan** [comment]
> No, this PR doesn't have any user impacts.

AH I see, it's just the linter :)

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
Yes, it's a lint check.

File: tests/pyfunc/test_virtualenv.py (hunk 1 above)
Comment (B-Step62) [reply tests/pyfunc/test_virtualenv.py:68]: I think this approach is a clever way to reduce unnecessary (and not fun) work.
Do you plan to replace them all in this PR or in a follow-up?

Comment (harupy) [reply tests/pyfunc/test_virtualenv.py:68]: Good question. In this PR, I'll only update the violating log_model calls in `tests`. I'll update the rest in a follow-up PR.

Comment (B-Step62) [reply tests/pyfunc/test_virtualenv.py:68]: SG!

**Review by B-Step62 — APPROVED** [review]

**Pull request merged**